custom_components/elasticsearch/es_privilege_check.py

```diff
@@ -33,10 +33,21 @@ def __init__(self, es_gateway: ElasticsearchGateway, config: dict = {}):
     async def enforce_privileges(self, config: dict = None):
         """Ensure client is configured with properly authorized credentials."""
         LOGGER.debug("Starting privilege enforcement")
-        result = await self.check_privileges(config if config else self.config)
-        if not result.has_all_requested:
-            LOGGER.debug("Required privileges are missing.")
-            raise InsufficientPrivileges()
+
+        # TODO: Figure out why this is needed
```

Comment:
A previous version of this file required the ability to override `self.config` with the local `config`. I can't remember why, but it was either for the config flow or unit testing.
That said, I can't find a reason why we can't consolidate this now, either by always using `self.config`, or always using local `config`. I don't have a strong preference which we choose. We can also clean this up in a followup PR.

Reply:
Some tests start failing if we only use self.config and I couldn't really figure out why but I didnt spend a ton of time on it